troposphere/emr.py

```diff
@@ -122,15 +122,76 @@ class EbsConfiguration(AWSProperty):
     }
 
 
+class Constraints(AWSProperty):
+    props = {
+        'MinCapacity': (integer, False),
+        'MaxCapacity': (integer, False)
+    }
+
+
+class CloudWatchAlarmDefinition(AWSProperty):
+    props = {
+        'ComparisonOperator': (basestring, False),
+        'EvaluationPeriods': (positive_integer, False),
+        'MetricName': (basestring, False),
+        'Namespace': (basestring, False),
+        'Period': (positive_integer, False),
+        'Statistic': (basestring, False),
+        'Threshold': (positive_integer, False),
+        'Unit': (basestring, False),
+        'Dimensions': ([KeyValue], False)
+    }
+
+
+class Trigger(AWSProperty):
+    props = {
+        'CloudWatchAlarmDefinition': (CloudWatchAlarmDefinition, False),
+    }
+
+
+class SimpleScalingPolicyConfiguration(AWSProperty):
+    props = {
+        'AdjustmentType': (basestring, False),
+        'ScalingAdjustment': (positive_integer, False),
+        'CoolDown': (positive_integer, False)
+    }
+
+
+class RulesActionConfig(AWSProperty):
+    props = {
+        'Market': (market_validator, False),
+        'SimpleScalingPolicyConfiguration': (
+            SimpleScalingPolicyConfiguration, False
+        )
+    }
+
+
+class Rules(AWSProperty):
+    props = {
+        'Name': (basestring, False),
+        'Description': (basestring, False),
+        'Action': (RulesActionConfig, False),
+        'Trigger': (Trigger, False)
+    }
+
+
+class AutoScalingPolicy(AWSProperty):
+    props = {
+        'Constraints': (Constraints, True),
+        'Rules': ([Rules], False)
+    }
+
+
 class InstanceGroupConfigProperty(AWSProperty):
     props = {
         'BidPrice': (basestring, False),
         'Configurations': ([Configuration], False),
         'EbsConfiguration': (EbsConfiguration, False),
-        'InstanceCount': (positive_integer, True),
+        'InstanceCount': (positive_integer, False),
```

Comment:
From the docs it looks like this is required: http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-emr-cluster-jobflowinstancesconfig-instancegroupconfig.html